dulwich/porcelain.py

```diff
@@ -1186,10 +1197,10 @@ def check_ignore(repo, paths, no_index=False):
         index = r.open_index()
         ignore_manager = IgnoreFilterManager.from_repo(r)
         for path in paths:
-            if os.path.isabs(path):
-                path = os.path.relpath(path, r.path)
             if not no_index and path_to_tree_path(r.path, path) in index:
                 continue
+            if os.path.isabs(path):
```

Comment:
I suspect this change is what made the python2.7 tests fail.

Reply:
thks, gonna check it out